rendering_resource_manager_service/session/management/session_manager.py

```diff
@@ -289,28 +291,35 @@ def query_status(session_id):
                 # serving REST requests. The vocabulary is invoked to make
                 # sure that the rendering resource is ready to serve REST
                 # requests.
-                log.info(1, 'Requesting rendering resource vocabulary')
-                status = SessionManager.request_vocabulary(session_id)
-                if status[0] == 200:
+                rr_settings = \
+                    manager.RenderingResourceSettingsManager.get_by_id(session.renderer_id.lower())
+                if not rr_settings.wait_until_running:
                     log.info(1, 'Rendering resource is now RUNNING!')
                     session.status = SESSION_STATUS_RUNNING
                     session.save()
                 else:
-                    if session.job_id == '':
-                        hostname = 'localhost'
-                        status_description = str(session.renderer_id + ' is starting... ')
+                    log.info(1, 'Requesting rendering resource vocabulary')
+                    status = SessionManager.request_vocabulary(session_id)
+                    if status[0] == 200:
```

Comment:
Changed in that file only. Will open dedicated PR for all the other files in the project.